torch/_inductor/graph.py

```diff
@@ -366,9 +367,10 @@ def find_nodes_prefer_channels_last(self):
         """
         output_set = set()
         for n in reversed(self.module.graph.nodes):
-            if n.target == torch.ops.aten.convolution.default:
-                output_set.add(n)
-                continue
+            if inductor_config.conv_force_channels_last:
+                if n.target == torch.ops.aten.convolution.default:
+                    output_set.add(n)
+                    continue
 
```

Comment:
We should be making channels-last occur here.


https://github.com/pytorch/pytorch/blob/0522fa5f9676dfe5eabadc78f83b8876145c5bf4/torch/_inductor/graph.py#L248-L257

Reply:
@pragupta was able to take the comment and submit a PR which is much cleaner using layout_opt could you take a look at https://github.com/pytorch/pytorch/pull/107812

This PR will now focus on adding the device_type into compile_meta and getting our ROCm compatible triton configs

I've also submitted an update to allow for conditional logic in the conv/mm configs which should avoid duplication, could get your review on this?